specs/specs.js

```diff
@@ -73,83 +83,106 @@ describe('Fingerprint2', function () {
+        var key = 'pixelRatio'
+        var fp2 = new Fingerprint2({excludes: [key]})
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === key
+          })).toBeFalse()
           done()
         })
       })
 
       it('does not use deviceMemory when excluded', function (done) {
-        var fp2 = new Fingerprint2({excludeDeviceMemory: true})
-        spyOn(fp2, 'getDeviceMemory')
-        fp2.get(function (result) {
-          expect(fp2.getDeviceMemory).not.toHaveBeenCalled()
+        var key = 'deviceMemory'
+        var fp2 = new Fingerprint2({excludes: [key]})
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === key
+          })).toBeFalse()
           done()
         })
       })
 
       it('does not use screen resolution when excluded', function (done) {
-        var fp2 = new Fingerprint2({excludeScreenResolution: true})
-        spyOn(fp2, 'getScreenResolution')
-        fp2.get(function (result) {
-          expect(fp2.getScreenResolution).not.toHaveBeenCalled()
+        var key = 'screenResolution'
+        var fp2 = new Fingerprint2({excludes: [key]})
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === key
+          })).toBeFalse()
           done()
         })
       })
 
       it('does not use available screen resolution when excluded', function (done) {
-        var fp2 = new Fingerprint2({excludeAvailableScreenResolution: true})
-        spyOn(fp2, 'getAvailableScreenResolution')
-        fp2.get(function (result) {
-          expect(fp2.getAvailableScreenResolution).not.toHaveBeenCalled()
+        var key = 'availableScreenResolution'
+        var fp2 = new Fingerprint2({excludes: [key]})
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === key
+          })).toBeFalse()
           done()
         })
       })
 
       it('does not use plugins info when excluded', function (done) {
-        var fp2 = new Fingerprint2({excludePlugins: true})
-        spyOn(fp2, 'getRegularPlugins')
-        fp2.get(function (result) {
-          expect(fp2.getRegularPlugins).not.toHaveBeenCalled()
+        var key = 'plugins'
+        var fp2 = new Fingerprint2({excludes: [key]})
+        fp2.get(function (result, components) {
+          expect(components.some(function (componentResult) {
+            return componentResult.key === key
+          })).toBeFalse()
           done()
         })
       })
 
-      it('does not use IE plugins info when excluded', function (done) {
+      // navigator.appName is read only
+      xit('does not use IE plugins info when excluded', function (done) {
```

Comment:
Do we have a non-excluded test for IE plugins? If not, we should a new issue to remember to add a positive test in the future.

Reply:
we should create issue then. Note that I disabled the test because it always passes because phantomjs is not detected as IE.